tests_typing/test_item_page.mypy-testing

```diff
@@ -0,0 +1,90 @@
+import attr
+import pytest
+import attrs
+
+from web_poet import ItemPage, field
+from web_poet.pages import ItemT
+
+
+@attrs.define
+class Item:
+    name: str
+
+
+@pytest.mark.mypy_testing
+@pytest.mark.xfail
+async def test_item_page() -> None:
+    class MyPage(ItemPage):
+        @field
+        def name(self):
+            return "hello"
+
+    page = MyPage()
+    item = await page.to_item()
+    reveal_type(item)  # R: dict
+
+
+@pytest.mark.mypy_testing
+async def test_item_page_parametrized() -> None:
+    class MyPage(ItemPage[Item]):
+        @field
+        def name(self):
+            return "hello"
+
+    page = MyPage()
+    item = await page.to_item()
+    reveal_type(item)  # R: __main__.Item
+
+
+@pytest.mark.mypy_testing
+async def test_item_page_parametrized_subclass() -> None:
+    class BasePage(ItemPage[Item]):
+        @field
+        def name(self):
+            return "hello"
+
+
+    class Subclass(BasePage):
+        pass
+
+    page = Subclass()
+    item = await page.to_item()
+    reveal_type(item)  # R: __main__.Item
+
+
+@pytest.mark.mypy_testing
+async def test_item_page_subclass_parametrized() -> None:
+    class BasePage(ItemPage[ItemT]):
+        @field
+        def name(self):
+            return "hello"
+
+    class Subclass(BasePage[Item]):
+        pass
+
+    page = Subclass()
+    item = await page.to_item()
+    reveal_type(item)  # R: __main__.Item
+
+
+@pytest.mark.mypy_testing
+@pytest.mark.xfail
+async def test_item_page_change_type() -> None:
+
+    class BasePage(ItemPage[Item]):
+        @field
+        def name(self):
+            return "hello"
+
+    @attr.define
+    class MyItem(Item):
+        price: float
+
+    class Subclass(BasePage, ItemPage[MyItem]):
+        @field
+        def price(self):
+            return 123
+
+    page = Subclass()
+    item = await page.to_item()
+    reveal_type(item)  # R: MyItem
```

Comment:
This typing failure is my main concern with this PR. For some reason (maybe for a valid one), mypy thinks the type of item is going to be Item, not MyItem; that's why a test case is marked as xfail. 

What's interesting is that Pycharm thinks it's MyItem, so IDE works properly. 

I haven't checked yet if this should be considered a mypy bug or not, it needs a bit more exploration.

Reply:
The alternative I tried (https://github.com/scrapinghub/web-poet/pull/75) looks worse, because typing of to_item() method results doesn't work at all there. Here it works at least in some cases.

Reply:
Interesting, it would seem **mypy** has still got some edge cases that are uncovered. I've tried looking deeper into this specific issue but to no avail. I'm wondering if we should also test this against other static type checkers like:
- https://github.com/microsoft/pyright
- https://github.com/facebook/pyre-check
- https://github.com/google/pytype

Reply:
I tried it with VS code + pylance, it agrees with mypy unfortunately. But I currently don't see a way around it, and alternatives look worse.

Reply:
I was also trying some of the workarounds mentioned in mypy's GH issue threads but couldn't make these particular examples work. Let's go with having them marked as xfail for now. 👍